### PR TITLE
Allow crate version deletion in the first 24 hours from creation

### LINKED ISSUE #5293: Allow deletion of crates within 24 hours of publishing
NPM has this policy to better handle accidental publishing.

### PR DESCRIPTION
**Pull request by nappa85**
Adds route DELETE /api/v1/crates/:crate_id

Resolves #5293

### CODE REVIEW

**Comment by Turbo87** [comment]
@nappa85 just for awareness: https://github.com/rust-lang/crates.io/issues/5293#issuecomment-1272406916

tl;dr I'm not sure whether we should allow crate deletion, or if it makes more sense for us to allow version deletion, and then if there is no version left the crate automatically gets deleted too.

**Comment by nappa85** [comment]
> @nappa85 just for awareness: [#5293 (comment)](https://github.com/rust-lang/crates.io/issues/5293#issuecomment-1272406916)
> 
> tl;dr I'm not sure whether we should allow crate deletion, or if it makes more sense for us to allow version deletion, and then if there is no version left the crate automatically gets deleted too.

Updated, let me know if there is anything else

**Comment by Turbo87** [comment]
I think there is some work left on correctly syncing both the git and the http index after the deletion. But before we continue, I talked to a few people about this and since it is a quite significant change we might actually require to go through the RFC process for it since other people are currently relying on the fact that crates don't just disappear like that again.

**Comment by Turbo87** [comment]
and just to be clear, I'm very thankful for the quick implementation! I just want to make sure we're not we're not surprising too many people with this change :)

**Comment by nappa85** [comment]
> and just to be clear, I'm very thankful for the quick implementation! I just want to make sure we're not we're not surprising too many people with this change :)

Understandable, don't worry

**Comment by nappa85** [comment]
Something nasty happened during the rebase, probably better restart from scratch...

**Comment by Turbo87** [comment]
FWIW there is now an RFC for this: https://github.com/rust-lang/rfcs/pull/3660

since the implementation will need a couple of changes (and rebases) I'll close this PR and will use it as inspiration when implementing the RFC spec :)
